bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/engine/impl/HttpCacheEngineImpl.java

```diff
@@ -1,77 +1,108 @@
+ * to this.
  */
-
-// TODO - Make this service tied to osgi config nodes
-// TODO - Remove the immediate false annotation.
-
-
 // @formatter:off
-@Component(immediate = true)
-@Service
-@References({@Reference(name = HttpCacheEngineImpl.METHOD_NAME_TO_BIND_CONFIG,
-                        referenceInterface = HttpCacheConfig.class,
-                        policy = ReferencePolicy.DYNAMIC,
-                        cardinality = ReferenceCardinality.OPTIONAL_MULTIPLE),
-
-                   /* @Reference(name = HttpCacheEngineImpl.METHOD_NAME_TO_BIND_CACHE_STORE,
-                               referenceInterface = HttpCacheStore.class,
-                               policy = ReferencePolicy.DYNAMIC,
-                               cardinality = ReferenceCardinality.MANDATORY_MULTIPLE),
-
-                    @Reference(name = HttpCacheEngineImpl.METHOD_NAME_TO_BIND_CACHE_HANDLING_RULES,
-                               referenceInterface = HttpCacheHandlingRule.class,
-                               policy = ReferencePolicy.DYNAMIC,
-                               cardinality = ReferenceCardinality.OPTIONAL_MULTIPLE)*/})
+@Component
+@Properties({
+        @Property(name = "jmx.objectname",
+                value = "com.adobe.acs.httpcache:type=HTTP Cache Engine",
+                propertyPrivate = true)
+})
+@References({
+        @Reference(name = HttpCacheEngineImpl.METHOD_NAME_TO_BIND_CONFIG,
+                referenceInterface = HttpCacheConfig.class,
+                policy = ReferencePolicy.DYNAMIC,
+                cardinality = ReferenceCardinality.MANDATORY_MULTIPLE),
+
+        @Reference(name = HttpCacheEngineImpl.METHOD_NAME_TO_BIND_CACHE_HANDLING_RULES,
+               referenceInterface = HttpCacheHandlingRule.class,
+               policy = ReferencePolicy.DYNAMIC,
+               cardinality = ReferenceCardinality.MANDATORY_MULTIPLE),
+
+        @Reference(name = HttpCacheEngineImpl.METHOD_NAME_TO_BIND_CACHE_STORE,
+               referenceInterface = HttpCacheStore.class,
+               policy = ReferencePolicy.DYNAMIC,
+               cardinality = ReferenceCardinality.MANDATORY_MULTIPLE)
+})
+@Service(value = {DynamicMBean.class, HttpCacheEngine.class})
 // @formatter:on
-public class HttpCacheEngineImpl implements HttpCacheEngine {
+public class HttpCacheEngineImpl extends AnnotatedStandardMBean implements HttpCacheEngine, HttpCacheEngineMBean {
     private static final Logger log = LoggerFactory.getLogger(HttpCacheConfigImpl.class);
 
     /** Method name that binds cache configs */
     static final String METHOD_NAME_TO_BIND_CONFIG = "httpCacheConfig";
 
     /** Thread safe list to contain the registered HttpCacheConfig references. */
-    private static CopyOnWriteArrayList<HttpCacheConfig> cacheConfigs = new CopyOnWriteArrayList<>();
+    private static final ConcurrentSkipListSet<HttpCacheConfig> cacheConfigs = new ConcurrentSkipListSet<>(new HttpCacheConfigComparator());
```

Comment:
@davidjgonzalez There is an inconsistency as we use ConcurrentSkipListSet for tracking cache config. ConcurrentSkipListSet uses the given Comparitor for comparing and checking uniqueness. With this current implementation, we would be able to have only one element for the given cache config order at any time. 

We may have to go back to CopyOnWriteArrayList/Set and take care of ordering in the HttpCacheConfig.getCacheConfig method. What do you think?